scripts/setuplib.py

```diff
@@ -302,24 +303,18 @@ def generate_all_cython():
     Run all of the cython that needs to be generated.
     """
 
-    threads = [ ]
-
-    for args in generate_cython_queue:
-
-        if "RENPY_CYTHON_SINGLETHREAD" in os.environ:
+    if "RENPY_CYTHON_SINGLETHREAD" in os.environ:
+        for args in generate_cython_queue:
             generate_cython(*args)
             if cython_failure:
                 sys.exit(1)
-        else:
-            t = threading.Thread(target=generate_cython, args=args)
-            t.start()
-            threads.append(t)
-
-    for t in threads:
-        t.join()
+    else:
+        with ThreadPoolExecutor() as executor:
+            for args in generate_cython_queue:
+                executor.submit(generate_cython, *args)
 
```

Comment:
Not certain, but do you not need a call to `shutdown` to await the threads to match the previous `join` behaviour?

Something like (untested):
```suggestion

            executor.shutdown(wait=True)
```

Reply:
No, shutdown() is called automatically when leaving the "with" block:

> You can avoid having to call this method explicitly if you use the [with](https://docs.python.org/3/reference/compound_stmts.html#with) statement, which will shutdown the [Executor](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Executor) (waiting as if [Executor.shutdown()](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Executor.shutdown) were called with wait set to True):

https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Executor.shutdown

Reply:
Ah, nice!